packages/dataviews/src/search.js

```diff
@@ -18,11 +18,13 @@ const Search = memo( function Search( { label, view, onChangeView } ) {
 		onChangeViewRef.current = onChangeView;
 	}, [ onChangeView ] );
 	useEffect( () => {
-		onChangeViewRef.current( {
-			...view,
-			page: 1,
-			search: debouncedSearch,
-		} );
+		if ( debouncedSearch !== view.search ) {
```

Comment:
Do we need to add `view` to the deps array?

Reply:
oops, yes.

Reply:
In the demo you shared, pagination is reset after layout switch. I don't know how you're using DataViews in your code, but happy to take a look if there's some PR out there. This is not something that happens in the core component:

https://github.com/WordPress/gutenberg/assets/583546/f3920de0-0e6a-43e4-a535-6dc5fd0c8eb7

Reply:
In pages we have preloaded the data (like Templates) this doesn't occur. In page though it does and it seems related to these lines of code: https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/page-pages/index.js#L65. Did you have similar code in your explorations @roo2 ? 

We need to check this either way. -cc @jorgefilipecosta

Reply:
@ntsekouras I think the reason for the issue in our code is that we are updating the URL to include the filter, pagination params e.g. `?page=2` etc. This is causing the dataviews component to be re-rendered, which causes the [line you mentioned]( https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/page-pages/index.js#L65) to be hit several times and then the line I changed gets hit which would have reset the pagination before applying my change.

Reply:
I can try to create a little demo or test case with re-rendering to prove this